src/middlewared/middlewared/plugins/device_/device_info_linux.py

```diff
@@ -253,23 +252,48 @@ def get_storage_devices_topology(self):
                 }
         return topology
 
-    async def get_gpus(self):
-
-        if self.GPU:
-            return self.GPU
-
-        not_available = {'available': False, 'vendor': None}
+    def get_gpus(self):
+        cp = subprocess.Popen(['lspci', '-D'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = cp.communicate()
+        if cp.returncode:
+            raise CallError(f'Unable to list available gpus: {stderr.decode()}')
+
```

Comment:
Why are we no longer caching the GPU information?

Reply:
We would need to clear the cache each time a user updated his gpu configuration, i am not sure if the management ( or keeping it in sync ) of keeping the cache fresh is better then just not using cache at all as this method is not going to be called very often.

Reply:
Are GPUs hot-pluggable? Not sure how a GPU configuration can be updated without shutting down the PC.

Reply:
They could be but we are not looking to support that. I am looking to make sure we have updated information being displayed to the user if a GPU can be used by host or not. So user changing configuration means user wanting to isolate a GPU from the host and we reflecting that here.

Reply:
If we're not going to support hot-pluggable GPUs, then caching this data seems to make more sense.

Reply:
Yes but keeping the cache in sync for one subprocess call which is also not really time/resource intensive ? Do you think  we should be doing that ?

Reply:
I have no issues doing it btw :p Just raising my concern, we can do that of course or maybe even handle it differently by updating `available_to_host` field each time on cache - that seems like a good compromise btw 🤔

Reply:
I was just going to say maybe put `GPUS = None` in the `DeviceService` class. And then update that variable with the `get_gpus` method if `GPUS = None` else just return `self.GPUS`.

Reply:
So that worked if the information did not change through out the uptime of the system, but that changes. Specifically `available_to_host` field changes.
I think a fair compromise is to update `self.GPUS` always with logic in place for `available_to_host` field and use cache.

Reply:
I didn't realize that attribute changes after boot (seems weird btw). In that case, probably best to just not cache like you originally did and move on. Not worth trying to solve cache validation/invalidation issues especially if this method isn't called often and isn't intensive.

Reply:
Thanks for explanation.